Sources/IO/Geometry/GLTFReader/Utils.js

```diff
@@ -0,0 +1,152 @@
+import vtkTexture from 'vtk.js/Sources/Rendering/Core/Texture';
+import macro from 'vtk.js/Sources/macros';
+import WebworkerPromise from 'webworker-promise';
+import Worker from './ORMTexture.worker';
+import { BYTES, COMPONENTS, ARRAY_TYPES, GL_SAMPLER } from './Contants';
+
+const { vtkWarningMacro } = macro;
+
+export function getBytesFromComponentType(componentType) {
+  return BYTES[componentType];
+}
+
+export function getSizeFromAccessorType(type) {
+  return COMPONENTS[type];
+}
+
+// When undefined, a sampler with repeat wrapping and auto filtering should be used.
+// https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#texture
+export function getGLEnumFromSamplerParameter(parameter) {
+  const GL_TEXTURE_MAG_FILTER = 0x2800;
+  const GL_TEXTURE_MIN_FILTER = 0x2801;
+  const GL_TEXTURE_WRAP_S = 0x2802;
+  const GL_TEXTURE_WRAP_T = 0x2803;
+
+  // Sampler default values
+  // const GL_REPEAT = 0x2901;
+  // const GL_RLINEAR = 0x2601;
+  // const GL_RNEAREST_MIPMAP_LINEAR = 0x2702;
+
+  const PARAMETER_MAP = {
+    magFilter: GL_TEXTURE_MAG_FILTER,
+    minFilter: GL_TEXTURE_MIN_FILTER,
+    wrapS: GL_TEXTURE_WRAP_S,
+    wrapT: GL_TEXTURE_WRAP_T,
+  };
+
+  return PARAMETER_MAP[parameter];
+}
+
+export function getAccessorArrayTypeAndLength(accessor, bufferView) {
+  const ArrayType = ARRAY_TYPES[accessor.componentType];
+  const components = COMPONENTS[accessor.type];
+  const bytesPerComponent = BYTES[accessor.componentType];
+  const length = accessor.count * components;
+  const byteLength = accessor.count * components * bytesPerComponent;
+  return { ArrayType, length, byteLength };
+}
+
+export function resolveUrl(url, orginalPath) {
```

Comment:
I haven't looked too closely to what's going on here, but could some of this be serviced using the `URL` API? In my opinion, it's better to favor URL parsers over regexes for readability (depending on circumstances, of course).